tchannel/tornado/dispatch.py

```diff
@@ -233,17 +234,31 @@ def _on_post_response(future):
             e.id = request.id
             connection.send_error(e)
         except Exception as e:
+            exc_type, exc_obj, exc_tb = sys.exc_info()
```

Comment:
Re-assigning shouldn't be a problem because then you're no longer holding onto the reference, but since you want to log the original *full* stack trace in `log.error`, you may need to hold onto the original reference walk the traceback in a separate variable.